File: flutter_app/lib/models/evaluation_record.dart
class EvaluationRecord {
  final int id;
  final String patientName;
  final String patientId;
  final double score;
  final String date;
  final int age;
  final String gender;
  final String boneQuality;
  final String jawLocation;
  final String smokingStatus;
  final String diabetes;
  final String periodontitis;
  final String bruxism;
  final String oralHygiene;
  final double implantLength;
  final double implantDiameter;
  final String implantSurface;
  final Map<String, dynamic> rawJson;

  EvaluationRecord({
    required this.id,
    required this.patientName,
    required this.patientId,
    required this.score,
    required this.date,
    required this.age,
    required this.gender,
    required this.boneQuality,
    required this.jawLocation,
    required this.smokingStatus,
    required this.diabetes,
    required this.periodontitis,
    required this.bruxism,
    required this.oralHygiene,
    required this.implantLength,
    required this.implantDiameter,
    required this.implantSurface,
    required this.rawJson,
  });

  factory EvaluationRecord.fromJson(Map<String, dynamic> json) {
    return EvaluationRecord(
      id: json['id'] ?? 0,
      patientName: json['patient_name'] ?? 'Unknown Patient',
      patientId: json['patient_id'] ?? 'PID-UNKNOWN',
      score: (json['score'] as num?)?.toDouble() ?? 50.0,
      date: json['date'] ?? 'Today',
      age: json['age'] ?? 45,
      gender: json['gender'] ?? 'Male',
      boneQuality: json['bone_quality'] ?? 'Type 2',
      jawLocation: json['jaw_location'] ?? 'Mandible',
      smokingStatus: json['smoking_status'] ?? 'Non-smoker',
      diabetes: json['diabetes'] ?? 'no',
      periodontitis: json['history_periodontitis'] ?? 'no',
      bruxism: json['bruxism'] ?? 'no',
      oralHygiene: json['oral_hygiene'] ?? 'Good',
      implantLength: (json['implant_length_mm'] as num?)?.toDouble() ?? 10.0,
      implantDiameter: (json['implant_diameter_mm'] as num?)?.toDouble() ?? 4.0,
      implantSurface: json['implant_surface'] ?? 'Roughened',
      rawJson: json,
    );
  }

  String get riskCategory {
    if (score >= 90.0) return 'Low Risk (Excellent Prognosis)';
    if (score >= 80.0) return 'Medium Risk (Moderate Prognosis)';
    return 'High Risk Profile';
  }
}

File: android_app/app/src/main/java/com/implantai/app/models/EvaluationRecord.java
package com.implantai.app.models;

import org.json.JSONObject;

public class EvaluationRecord {
    private int id;
    private String patientName;
    private String patientId;
    private double score;
    private String date;
    private int age;
    private String gender;
    private String boneQuality;
    private String jawLocation;
    private String smokingStatus;
    private String diabetes;
    private String periodontitis;
    private String bruxism;
    private String oralHygiene;
    private double implantLength;
    private double implantDiameter;
    private String implantSurface;
    private JSONObject rawJson;

    public EvaluationRecord(JSONObject obj) {
        this.rawJson = obj;
        this.id = obj.optInt("id", 0);
        this.patientName = obj.optString("patient_name", "Unknown Patient");
        this.patientId = obj.optString("patient_id", "PID-UNKNOWN");
        this.score = obj.optDouble("score", 50.0);
        this.date = obj.optString("date", "Today");
        this.age = obj.optInt("age", 45);
        this.gender = obj.optString("gender", "Male");
        this.boneQuality = obj.optString("bone_quality", "Type 2");
        this.jawLocation = obj.optString("jaw_location", "Mandible");
        this.smokingStatus = obj.optString("smoking_status", "Non-smoker");
        this.diabetes = obj.optString("diabetes", "no");
        this.periodontitis = obj.optString("history_periodontitis", "no");
        this.bruxism = obj.optString("bruxism", "no");
        this.oralHygiene = obj.optString("oral_hygiene", "Good");
        this.implantLength = obj.optDouble("implant_length_mm", 10.0);
        this.implantDiameter = obj.optDouble("implant_diameter_mm", 4.0);
        this.implantSurface = obj.optString("implant_surface", "Roughened");
    }

    public int getId() { return id; }
    public String getPatientName() { return patientName; }
    public String getPatientId() { return patientId; }
    public double getScore() { return score; }
    public String getDate() { return date; }
    public int getAge() { return age; }
    public String getGender() { return gender; }
    public String getBoneQuality() { return boneQuality; }
    public String getJawLocation() { return jawLocation; }
    public String getSmokingStatus() { return smokingStatus; }
    public String getDiabetes() { return diabetes; }
    public String getPeriodontitis() { return periodontitis; }
    public String getBruxism() { return bruxism; }
    public String getOralHygiene() { return oralHygiene; }
    public double getImplantLength() { return implantLength; }
    public double getImplantDiameter() { return implantDiameter; }
    public String getImplantSurface() { return implantSurface; }
    public JSONObject getRawJson() { return rawJson; }

    public String getRiskCategory() {
        if (score >= 90.0) return "Low Risk (Excellent Prognosis)";
        if (score >= 80.0) return "Medium Risk (Moderate Prognosis)";
        return "High Risk Profile";
    }

    public int getRiskColor() {
        if (score >= 90.0) return 0xFF27AE60;
        if (score >= 80.0) return 0xFFE67E22;
        return 0xFFC0392B;
    }
}
